README.md

```diff
@@ -17,49 +17,205 @@ information](LICENSE), as well as [pre-commit hooks](https://pre-commit.com)
+
+## Image tags ##
+
+The images of this container are tagged with [semantic
+versions](https://semver.org) of the underlying example project that they
+containerize.  It is recommended that most users use a version tag.  e.g.,
+`:0.0.1`
+
+| Image:tag | Description |
+|-----------|-------------|
+|`cisagov/example:1.2.3`| An exact release version. |
+|`cisagov/example:1.2`| The most recent release matching the major and minor version numbers. |
+|`cisagov/example:1`| The most recent release matching the major version number. |
+|`cisagov/example:edge` | The most recent image built from a merge into the `develop` branch of this repository. |
+|`cisagov/example:nightly` | A nightly build of the `develop` branch of this repository. |
+|`cisagov/example:latest`| The most recent release image pushed to a container registry.  Pulling an image using the `:latest` tag [should be avoided.](https://vsupalov.com/docker-latest-tag/) |
+
+See the [tags tab](https://hub.docker.com/r/cisagov/example/tags) on Docker
+Hub for a list of all the supported tags.
 
 ## Volumes ##
 
 | Mount point | Purpose        |
 |-------------|----------------|
-| /var/log    | logging output |
+| `/var/log`  |  Log storage   |
+
+## Environment variables ##
+
+### Required ###
+
+There are no required environment variables.
+
+<!--
+| Name  | Purpose | Default |
+|-------|---------|---------|
+| `REQUIRED_VARIABLE` | Describe its purpose. | `null` |
+-->
+
+### Optional ###
+
+| Name  | Purpose | Default |
+|-------|---------|---------|
+| `ECHO_MESSAGE` | Sets the message echoed by this container.  | `Hello World from Dockerfile` |
+
+## Secrets ##
+
+| Filename     | Purpose |
+|--------------|---------|
+| `quote.txt` | Replaces the secret stored in the example library's package data. |
+
+## Building from source ##
+
+Build the image locally using this git repository as the [build context](https://docs.docker.com/engine/reference/commandline/build/#git-repositories):
+
+```console
+docker build \
+  --build-arg VERSION=0.0.1 \
+  --tag cisagov/example:0.0.1 \
+  https://github.com/cisagov/example.git#develop
+```
+
+## Cross-platform builds ##
+
+To create images that are compatible with other platforms you can use the
+[`buildx`](https://docs.docker.com/buildx/working-with-buildx/) feature of
+Docker:
+
+1. Copy the project to your machine using the `Code` button above
+   or the command line:
+
+    ```console
+    git clone https://github.com/cisagov/example.git
+    cd example
+    ```
+
+1. Create the `Dockerfile-x` file with `buildx` platform support:
+
+    ```console
+    ./buildx-dockerfile.sh
+    ```
+
+1. Build the image using `buildx`:
+
+    ```console
+    docker buildx build \
+      --file Dockerfile-x \
+      --platform linux/amd64 \
+      --build-arg VERSION=0.0.1 \
+      --output type=docker \
+      --tag cisagov/example:0.0.1 .
+    ```
 
 ## New Repositories from a Skeleton ##
```

Comment:
```suggestion
## New repositories from a skeleton ##
```

Reply:
I would like this fixed [upstream](https://github.com/cisagov/skeleton-generic/blob/develop/README.md#new-repositories-from-a-skeleton) instead.